.github/workflows/docker_kms_sgx.yml

```diff
@@ -0,0 +1,65 @@
+---
+name: Docker
+
+on:
+  workflow_call:
+
+env:
+  REGISTRY: ghcr.io
+  REGISTRY_IMAGE: ghcr.io/cosmian/enclave-kms-insecure
+
+jobs:
+  build-and-push-image:
+    runs-on: [self-hosted, sgx]
+    defaults:
+      run:
+        working-directory: sgx_ci
+    container: docker:19.03.15
+
+    steps:
+
+      - uses: actions/checkout@v1
+
+      - name: Login to GitHub Packages
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract Docker metadata
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: |
+            ${{ env.REGISTRY_IMAGE }}
+          tags: |
+            type=ref,event=branch
+            type=ref,event=pr
+            type=semver,pattern={{version}}
+            type=semver,pattern={{major}}.{{minor}}
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Build and tag docker container
+        uses: docker/build-push-action@v3
+        with:
+          file: ./sgx_ci/Dockerfile.sgx
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+          network: host
+          build-args: |
+            FEATURES=--features staging
```

Comment:
Could you add a comment here for the next reader explaining why we use this feature?